### PR TITLE
Fixed single feeds load

### PR DESCRIPTION
**Pull request by yatishTrootech**
APIS to be tested 

###   With or without staff user
CRUD means (List, Detail, create, update, delete)
1. comments CRUD
2. appreciate CRUD
3. Poll Vote
4. Flag 
5. Pin post 
6. Feedback CRUD
7. posts CRUD

### CODE REVIEW

File: feeds/views.py [hunk 1]
```diff
@@ -592,9 +585,10 @@ def appreciated_by(self, request, *args, **kwargs):
         if not post_id:
             raise ValidationError(_('Post ID required to appreciate a post'))
         post_id = int(post_id)
-        accessible_posts = accessible_posts_by_user(user, organization, False, False, post_id). \
-            values_list('id', flat=True)
-        if post_id not in accessible_posts:
+        try:
+            accessible_posts = accessible_posts_by_user(user, organization, False, False, post_id)
+            accessible_posts.distinct().get(id=post_id)
```
Comment (rishireward) [inline feeds/views.py:590]: distinct will take time, I would say simply check filter().exists()

File: feeds/views.py [hunk 2]
```diff
@@ -613,12 +607,13 @@ def answers(self, request, *args, **kwargs):
...
             raise ValidationError(_('Post ID required to retrieve all the related answers'))
         post_id = int(post_id)
-        accessible_posts = accessible_posts_by_user(user, organization, False, False, post_id).values_list('id', flat=True)
-        accessible_polls = accessible_posts.filter(post_type=POST_TYPE.USER_CREATED_POLL)
-        if post_id not in accessible_polls:
-            raise ValidationError(_('This is not a poll.'))
-        if post_id not in accessible_posts:
-            raise ValidationError(_('You do not have access to check the answers to this poll'))
+        try:
+            accessible_posts_by_user(user, organization, False, False, post_id).distinct().get(
```
Comment (rishireward) [inline feeds/views.py:611]: same here

File: feeds/views.py [hunk 3]
```diff
@@ -644,20 +639,13 @@ def vote(self, request, *args, **kwargs):
...
-        accessible_polls = accessible_posts.filter(
-            post_type=POST_TYPE.USER_CREATED_POLL
-        )
-        if post_id not in accessible_polls:
-            raise ValidationError(_('This is not a poll question'))
-        poll = None
         try:
-            poll = Post.objects.get(id=post_id)
-            poll.vote(user, answer_id)
+            poll = accessible_posts_by_user(user, organization, False, False, post_id).distinct().get(
```
Comment (rishireward) [inline feeds/views.py:643]: same here

File: feeds/views.py (hunk 1 above)
Comment (yatishTrootech) [reply feeds/views.py]: with distinct it is actually faster @rishireward 
image
image

**Review by rishireward — APPROVED** [review]

File: feeds/views.py (hunk 1 above)
Comment (dryice) [reply feeds/views.py]: can you do `with_distinct() - without_distinct()` @yatishTrootech ? Probably there's a DB level cache affecting the result.

Comment (yatishTrootech) [reply feeds/views.py]: there are possible chances that we may get duplicate results in the QS

**Review by rishireward — APPROVED** [review]

**Pull request merged**